### ISSUE TITLE
Replace SQL setup onConflictReplace with onConflictAbort

### ISSUE THREAD

**Issue by ataulm** [novoda/download-manager#218]
SQL setup allows onConflictReplace which can lead to unexpected state.

We should guard against conflicts in the first place so our db is never in an unexpected state.

Two tables:

- request
- download

It's already handled for file local URIs. It currently throws an `IllegalArgumentException` in two places:

- at the point of making the request - `DownloadRequest.Builder.withFile()` to make sure a single request doesn't contain files that will overwrite each other
- `DatabaseInteraction.submitRequest()` that makes sure we don't have any records in the database already where the `file_local_uri` matches the one in the request being submitted.

**Comment by ouchadam** [comment]
`SubmitRequest`  should throw or return an error result allowing the client to attempt to handle the error.

**Comment by ataulm** [comment]
Yep, atm it throws IllegalArgumentException but we should use a better name and apply it for the other cases.
